lib/src/main/kotlin/starknet/data/types/transactions.kt

```diff
@@ -24,12 +26,38 @@ enum class StarknetChainId(val value: Felt) {
     TESTNET(Felt.fromHex("0x534e5f474f45524c49")), // encodeShortString('SN_GOERLI'),
 }
 
-data class Invocation(
-    val contractAddress: Felt, val entrypoint: Felt, val calldata: Calldata?, val signature: Signature?
-)
+enum class BlockTag(val tag: String) {
+    LATEST("latest"),
+    PENDING("pending")
+}
 
-data class InvocationDetails(
-    val nonce: Felt?, val maxFee: Felt?, val version: Felt?
+@Serializable
+sealed class BlockHashOrTag() {
```

Comment:
Using this with block tag requires java user to do 
```
new BlockHashOrTag.Tag(BlockTag.PENDING)
new BlockHashOrTag.Hash(new Felt(X))
```
What if we did something like this?
```
sealed class BlockHashOrTag() {
    data class Hash(
        val blockHash: Felt
    ) : BlockHashOrTag() {
        override fun string(): String {
            return blockHash.hexString()
        }
    }

    sealed class BlockTag: BlockHashOrTag() {
        abstract val tag: String
        override fun string(): String {
            return tag
        }

        object LatestBlock : BlockTag() {
            override val tag = "latest"
        }

        object PendingBlock : BlockTag() {
            override val tag = "pending"
        }

        companion object {
            @field:JvmField
            public val LATEST_BLOCK = LatestBlock

            @field:JvmField
            public val PENDING_BLOCK = LatestBlock
        }

    }


    abstract fun string(): String
}
```

It behaves like an enum, is type safe and doesn't require specifying tag twice.

Reply:
The reason I wanted to define BlockTag outside of BlockHashOrTag class was to be able to use it in BlockNumOrTag as well, which will be needed for some endpoints